doc/rm/comportability_specification/index.md

```diff
@@ -440,18 +440,70 @@ Unlike interrupts, there is no software component to alerts at the peripheral, t
 See that [specification]({{< relref "/hw/ip/alert_handler/doc" >}}) for full details.
 A general description of the handling of alerts at the hardware level is given here.
 
-### Alerts per module
+### Alerts per Module
 
 Alerts are sent as a bundled output from a peripheral to the hardware alert handler.
 Each peripheral can send zero or more alerts, where each is a distinguishable security threat.
 Each alert originates in some internal event, and must be specially handled within the peripheral, and then within the alert handler module.
 
+Alerts of comportable IPs in the system must be in either of the following two categories:
+
+1. *Recoverable*, one-time triggered alerts.
+This category is for regular alerts that are due to recoverable error conditions.
+The alert sender transmits one single alert event when the corresponding error condition is asserted.
+
+2. *Fatal* alerts that are continuously triggered until reset.
+This category is for highly critical alerts that are due to terminal error conditions.
+The alert sender continuously transmits alert events until the system is reset.
+
+It is recommended that fatal alerts also trigger local security countermeasures, if they exist.
+For example, a redundantly encoded FSM that is glitched into an invalid state is typically considered to be a fatal error condition.
+In this case, a local countermeasure could be to move the FSM into a terminal error state in order to render the FSM inoperable until the next reset.
+
+The table below lists a few common error conditions and the recommended alert type for each of those errors.
+
+Error Event                                                | Regular IRQ | Recoverable Alert | Fatal Alert
+-----------------------------------------------------------|-------------|-------------------|-------------
+ECC correctable in NVM (OTP, Flash)                        | x           | x                 |
+ECC uncorrectable in Flash                                 | x           | x                 |
+ECC uncorrectable in OTP                                   | (x)         |                   | x
+Any ECC / Parity error in SRAMs or register files          | (x)         |                   | x
+Glitch detectors (e.g., invalid FSM encoding)              | (x)         |                   | x
+Incorrect usage of security IP (e.g., AES config register) | x           | x                 |
+Incorrect usage of regular IP                              | x           |                   |
+
+(x): optional
+
+The column "Regular IRQ" indicates whether the corresponding error condition should also send out a regular IRQ.
+Fatal alerts are not required to send out regular IRQs, since they cannot be recovered via SW actions.
```

Comment:
"are not required" implies that fatal alerts can still be routed as regular IRQs. Given that the chip will be forcefully reset imminently, are there any actual / valid usecases for wiring them as IRQs?

Reply:
OTBN always issues an `done` interrupt in addition to an alert to ensure that software can make forward progress even if specific alerts are disabled in the alert manager, causing no reset to be issued.

Reply:
I think Sri is trying to say that he can't see any use case for adding an IRQ on fatal alert. (Assuming I've got that right,) I think I probably agree.

Reply:
Right, my question was specific to fatal alerts.

Reply:
So the reason I still had these in the table as optional is that some IPs like OTP just trigger a regular IRQ whenever an error occurs (no matter whether fatal or recoverable).

However, given that this is our recommendation going forward, I do agree that there is no valid usecase for fatal alerts being routed as regular IRQs, and we can drop that from the table.

Reply:
I was referring to fatal alerts as well. In OTBN, we always issue an interrupt (the `done` interrupt, indicating that a processing operation has completed), *in addition* to any kind of alert (fatal or recoverable). The motivation is as follows: alerts (even fatal ones) are not guaranteed to reach the processor; they could be ignored/silenced in the alert manager. This is unlikely to be the case in production, but I don't find it unlikely to happen in development mode, or as workaround for a "broken" alert implementation in production. In such a case, we want the host (Ibex) software to be able to make progress, i.e. continue as if OTBN finished its operation (with an error). In case of a fatal alert, OTBN will of course lock up permanently, but the rest of the system should continue to operate fine until the alert manager reacts (with a system reset).

So IMO, we should
- either guarantee that a fatal alert results in a system reset
- or produce a design which can deal with "ignored" fatal alerts as we do in OTBN, which in OTBN's case means sending the `done` interrupt *in addition* to any kind of alert.

Reply:
Ok, I see. Given that we cannot guarantee that fatal alerts always result in a system reset, how about we keep the error IRQ as optional for all types alerts in the table?
The decision whether an IRQ is needed or not depends a lot on the programming model of a specific IP, hence I do not think it makes sense to mandate it.

Reply:
:+1:

Reply:
Amended, PTAL (last fixup commit).

Reply:
Thanks for the explanation @imphil. The amended wording LGTM.